spec/index.adoc

```diff
@@ -1674,6 +1674,96 @@ Any user can top-up any canister this way.
 
 This method is only available in local development instances.
 
+
+[#ic-bitcoin-canisters]
+== The IC bitcoin canisters
+
+Similar to the <<ic-management-canister,management canister>>, the Bitcoin functionality is also exposed through a facade, which we refer to as the "Bitcoin canisters".
+There are two Bitcoin canisters differing only in that one operates on the real Bitcoin blockchain whereas the other only works with the Bitcoin testnet.
+The Bitcoin canisters can be accessed using their addresses:
+
+- The canister address of the Bitcoin canister that uses the real Bitcoin blockchain is `TBD`.
+- The canister address of the Bitcoin canister that uses the Bitcoin testnet is `TBD`.
+
+It is *not* possible to interact with either Bitcoin canister via external requests (i.e., ingress messages).
+
+Information about Bitcoin can be found in the https://developer.bitcoin.org/devguide/[Bitcoin developer guides].
+
+[#ic-bitcoin-candid]
+=== Interface overview
+
+The following interface description, in https://github.com/dfinity/candid/blob/master/spec/Candid.md[Candid syntax], describes the available functionality. You can also link:{attachmentsdir}/bitcoin.did[download the file].
+----
+include::{example}bitcoin.did[]
+----
+
+The binary encoding of arguments and results are as per Candid specification.
+
+[#ic-bitcoin_get_utxos]
+=== IC method `get_utxos`
+
+Given a https://en.bitcoin.it/wiki/Base58Check_encoding[base58-encoded] address as part of the
+`GetUtxosRequest` , the function returns all unspent transaction outputs (UTXOs) associated with the
+provided address based on the current view of the Bitcoin blockchain available to the Bitcoin canister.
+
+If the call fails, e.g., because the address is malformed, a `GetUtxosError` variant is returned,
+indicating the reason for the failed call.
+
+The optional `min_confirmations` parameter can be used to limit the returned UTXOs to those with at
```

Comment:
Is there a maximum `min_confirmations` that can be used? I assume the upper bound is our `delta` parameter correct?

Reply:
Technically, `min_confirmations` can be as large as the length of the whole chain since we know the confirmation count for all outputs in the UTXO set.
If `min_confirmations` is larger than the length of the whole chain, an empty set is returned.